cpp/src/lists/interleave_columns.cu

```diff
@@ -172,29 +172,24 @@ struct interleave_list_entries_fn {
     rmm::mr::device_memory_resource* mr) const noexcept
   {
     auto const table_dv_ptr = table_device_view::create(input);
-    auto const comp_fn      = compute_string_sizes_and_interleave_lists_fn{
+    auto comp_fn            = compute_string_sizes_and_interleave_lists_fn{
       *table_dv_ptr, output_list_offsets.template begin<offset_type>(), data_has_null_mask};
 
-    if (data_has_null_mask) {
-      auto [offsets_column, chars_column, null_mask, null_count] =
-        cudf::strings::detail::make_strings_children_with_null_mask(
-          comp_fn, num_output_lists, num_output_entries, stream, mr);
-      return make_strings_column(num_output_entries,
-                                 std::move(offsets_column),
-                                 std::move(chars_column),
-                                 null_count,
-                                 std::move(null_mask),
-                                 stream,
-                                 mr);
-    }
+    auto validities =
+      rmm::device_uvector<int8_t>(data_has_null_mask ? num_output_entries : 0, stream);
+    comp_fn.d_validities = validities.data();
```

Comment:
I have concerns about what happens when `data_has_null_mask` is false. The pointer returned by `validities.data()` can't be used or compared against `nullptr`. 

Are we sure in that case that `comp_fn.d_validities` isn't used at all?

Reply:
Yes. The previous implementation already handled that case
This is just moving `validities` allocate from the now-removed utility to here.
https://github.com/rapidsai/cudf/pull/8830/files#diff-8d4b25b181461a330b69fb2f2dc66313e81b08c3dc7a45b2aa2560f50629c9c1L243

Reply:
Didn't the previous version only call the function when `data_has_null_mask` resolved to true, and now we call the new function even when that is false?

Reply:
Yes, but both paths used the same functor `comp_fn`.